### ISSUE TITLE
[Bug] _reconstruct_from_deprecated_options creates incorrect executor

### ISSUE THREAD

**Issue by mehrdadh** [apache/tvm#12409]
example:
```
target = tvm.target.Target("llvm -link-params=1")
executor = Executor("aot")
```
This will result in this executor: `graph{"link-params": 1}`

I looked at the logic of `_reconstruct_from_deprecated_options`, the issue is that if the executor config is not part of the target it will assume `graph` by default. This is obviously incorrect.

Possible solutions
- Pass `executor` to `_reconstruct_from_deprecated_options` and derive the final executor considering the original executor that was passed to build.
- Remove this function and instead show runtime error when wrong config is passed.

I suggest the second approach because the first approach is fairly complicated considering the number of configs and most likely it will result in wrong combination. This could be really annoying considering the fact that you could end up building with incorrect executor without noticing.

thoughts?

cc @areusch @Mousius

**Comment by mehrdadh** [comment]
cc @masahi

**Comment by Mousius** [comment]
Argh, yip, I think there's a few other permutations here - given we marked it as deprecated in `0.9` I think we should aim to remove it entirely in `0.10` so as to remove the confusion entirely?

**Comment by mehrdadh** [comment]
@Mousius I agree. Happy to send a PR and remove it. I think as a part of the refactor we should also remove `link-params` from target kinds like LLVM, C and hexagon.

**Comment by areusch** [comment]
yeah i also support removing those options from Target now that we've released 0.9.

**Comment by mehrdadh** [comment]
coo, I will send a PR
